DEVELOPMENT.md

```diff
@@ -42,23 +42,48 @@ cd flake8-bugbear
 /path/to/venv/bin/pip install -e '.[dev]'
 ```
 
+## Writing Tests
+
+flake8-bugbear has a test runner that will go through all files in `tests/eval_files/`, run them through the linter, and check that they emit the appropriate error messages.
+
+The expected errors are specified by adding comments on the line where the error is expected, using the format `# <error_code>: <col_offset>[, <var1>][, <var2>][...]`. E.g.
+```python
+x = ++n  # B002: 4
+try:
+    ...
+except* (ValueError,):  # B013: 0, "ValueError", "*"
+    ...
+```
+The error code should be in the `error_codes` dict, and the other values are passed to `eval` so should be valid python objects.
```

Comment:
I don't understand what the var1, var2, ... do, could you clarify?

> the other values are passed to `eval` so should be valid python objects

The example uses `"*"` which fails eval.

Reply:
Tried to clarify it, is it better now?

> The example uses "*" which fails eval.

well, `eval('"*"')` is fully valid. But I tried making that more explicit as well.

Feel free to suggest any changes if it still doesn't make sense, the relevant logic is

https://github.com/PyCQA/flake8-bugbear/blob/f7d5123f7aaefe27b9a1e8bbdf76b26157dfc336/tests/test_bugbear.py#L83-L92